### DISCUSSION TITLE
Плейлист со всеми треками одного артиста

### DISCUSSION THREAD

**Discussion by koojichan** [Chimildic/goofy#104] [Помощь с алгоритмом] [answered]
Я новичок в программировании, что-то да в этом понимаю, но!
Посидел несколько часиков, насоздавал плейлистов по шаблонам, изучил их принцип работы.
Решил создать чуть другое, а именно плейлист со всеми треками одного артиста, но какой раз меня ожидает неудача.
Вопрос следующий: возможно ли это вообще?

**Comment by Chimildic** [comment, marked as answer]
Наиболее простой вариант через [getArtistsTracks](https://chimildic.github.io/goofy/#/func?id=getartiststracks). В примере для исполнителя `nothing,nowhere`. Включая как альбомы, так и синглы. Остается изменить _id_ исполнителя на интересующее. 
```js
function createPlaylistForArtist() {
  let tracks = Source.getArtistsTracks({
    artist: {
      followed_include: false,
      include: [
        { id: '7FngGIEGgN3Iwauw1MvO4P' }, // nothing,nowhere
      ]
    },
    album: {
      groups: 'album,single',
    },
  });

  Playlist.saveWithReplace({
    // id: 'id',
    name: 'Все треки nothing,nowhere',
    tracks: tracks,
  })
}
```

**Reply by koojichan** [reply]
Спасибо огромное, там правда точки с запятой нет в самом конце.
Но тем не менее большое спасибо.
Я пытался ещё больше код написать, а тут не очень много строк понадобилось))

**Reply by Chimildic** [reply]
Точка с запятой в конце строки необязательна, но желательна для Javascript. Чтобы избежать сложных ошибок. В данном случае не она и вряд ли вообще встретится. Можно найти много примеров и обсуждений ([раз](https://learn.javascript.ru/structure#semicolon) , [два](https://ru.stackoverflow.com/questions/834135/%D0%91%D1%83%D0%B4%D1%83%D1%82-%D0%BB%D0%B8-%D0%BF%D1%80%D0%BE%D0%B1%D0%BB%D0%B5%D0%BC%D1%8B-%D0%B5%D1%81%D0%BB%D0%B8-%D0%BD%D0%B5-%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D1%8C-%D1%82%D0%BE%D1%87%D0%BA%D1%83-%D1%81-%D0%B7%D0%B0%D0%BF%D1%8F%D1%82%D0%BE%D0%B9-%D0%B2-javascript) и тд).
